File: hanoi.js
const hanoi2 = (discos, inicio, aux, destino) => {
  if (discos === 1) {
    console.log(`Move disco ${discos} de ${inicio} para ${destino}`)
  } else {
    hanoi2(discos - 1, inicio, destino, aux)
    console.log(`Move disco ${discos} de ${inicio} para ${destino}`)
    hanoi2(discos - 1, aux, inicio, destino)
  }
}

console.log('----')
hanoi2(30, 'Origem', 'Auxiliar', 'Destino')

File: Hard/hanoi.py
#-*-coding:cp1252-*-

def torres_hanoi(n,a,b,c):
  "Implementação das torres de Hanói"
  if n==1:
    print "move disco %d de %s para %s" % (n,a,c)
  else:
    torres_hanoi(n-1,a,c,b)
    print "Move disco %d de %s para %s" % (n,a,c)
    torres_hanoi(n-1,b,a,c)

print torres_hanoi(4,a='Início',b='Auxiliar',c='Fim')
